types/src/blockchain.rs

```diff
@@ -103,6 +106,40 @@ pub enum BlockchainReadRequest {
 
     /// A request for all alt blocks in the chain with the given [`ChainId`].
     AltBlocksInChain(ChainId),
+
+    /// Get a [`Block`] by its height.
+    Block {
+        height: usize,
+    },
+
+    /// Get a [`Block`] by its hash.
+    BlockByHash([u8; 32]),
+
+    /// Get the total amount of non-coinbase transactions in the chain.
+    TotalTxCount,
+
+    /// Get the current size of the database.
+    DatabaseSize,
+
+    // Get the difficulty for the next block in the chain.
+    Difficulty(usize),
+
+    /// Get an output histogram.
+    ///
+    /// TODO: document fields after impl.
+    OutputHistogram(OutputHistogramInput),
+
+    /// Get the coinbase amount and the fees amount for
+    /// `N` last blocks starting at particular height.
+    ///
+    /// TODO: document fields after impl.
+    CoinbaseTxSum {
+        height: usize,
+        count: u64,
+    },
+
+    /// Get the necessary data to create a custom block template.
+    MinerData,
```

Comment:
### Signatures (blockchain request)
Added to the existing `BlockchainReadRequest`.

Reply:
`Difficulty` is contained in the blockchain context so shouldn't need a new request (my bad for missing this).

`MinerData` will not be able to be a single request. Looking at its data it would need a request to the blockchain cache and to the tx-pool, both of these requests are already there/in this PR so this could just be removed I think